nixos/modules/services/networking/ntp/ntpd-rs.nix

```
@@ -0,0 +1,60 @@
+{ lib, config, pkgs, ... }:
+
+let
+  cfg = config.services.ntpd-rs;
+  configFile = pkgs.writers.writeTOML "ntpd-rs.toml" cfg.config;
```

We usually use the tools from `pkgs.formats` when specifying freeform settings like this. These freeform settings are usually also named `settings`, as specified in [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md) (remember to edit this in the tests as well). I'll add some related comments further down.

```suggestion
  format = pkgs.formats.toml { };
  configFile = format.generate "ntpd-rs.toml" cfg.settings;
```

---

Oh, yes, you're correct, thanks!